src/theory/strings/extf_solver.cpp

```diff
@@ -620,7 +621,8 @@ void ExtfSolver::checkExtfInference(Node n,
   if (inferEqr.getKind() == EQUAL)
   {
     // try to use the extended rewriter for equalities
-    inferEqrr = SequencesRewriter::rewriteEqualityExt(inferEqr);
+    inferEqrr =
+        StringsRewriter(&d_statistics.d_rewrites).rewriteEqualityExt(inferEqr);
```

Comment:
Can we actually just pass the instance of the rewriter here as a reference to the constructor instead of creating a new instance?